mautic/mautic
# Option to select domain for saml_idp_entity_id

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | No
| New feature/enhancement? (use the a.x branch)      | Yes
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | Yes 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 



#### Description:
User should be able to select between custom domain and subdomain to point to saml_idp_entity_id and configure their saml implementation accordingly.


#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. If we have 2 domains for instance like custom domain and subdomain then add subdomain in config file with key **subdomain_url**
3. set up saml sso using the custom domain
4. change the custom domain
5. update sso configuration to work with new custom domain
6. ensure sso breaks
7. change the saml_idp_entity_id using selectbox to match the desired value
8. ensure sso works as expected
